crates/core/tedge_mapper/src/lib.rs

```diff
@@ -140,16 +141,14 @@ pub async fn run(mapper_opt: MapperOpt) -> anyhow::Result<()> {
     let mapper_name = mapper_opt.name.to_string();
     let component = lookup_component(mapper_opt.name);
 
-    let tedge_config_location =
-        tedge_config::TEdgeConfigLocation::from_custom_root(&mapper_opt.common.config_dir);
-    let config = tedge_config::TEdgeConfig::try_new(tedge_config_location.clone()).await?;
-
     log_init(
         "tedge-mapper",
         &mapper_opt.common.log_args,
-        &tedge_config_location.tedge_config_root_path,
+        &mapper_opt.common.config_dir,
     )?;
 
+    let config = TEdgeConfig::load(&mapper_opt.common.config_dir).await?;
```

Comment:
The config is read a second time here. It could be passed as an argument to `run()`.

Reply:
Fixed: https://github.com/thin-edge/thin-edge.io/pull/3625/commits/3d32c78b31d78607dcf9854075f885b48006abd6